### PR TITLE
Feature/fix internal error handling

### PR DESCRIPTION
**Pull request by czardoz**

### CODE REVIEW

File: lib/runner/extensions/event.command.js [hunk 1]
```diff
@@ -91,9 +94,12 @@ module.exports = {
                     payload.trackContext.filter(function (variable) {
                         return _.isObject(payload.context[variable]);
                     }),
+                trackFailuresIn = _.isString(payload.trackFailuresIn) && payload.trackFailuresIn,
+                abortOnFailure = _.has(payload, 'abortOnFailure') ? payload.abortOnFailure : this.options.abortOnFailure,
                 stopOnScriptError = _.has(payload, 'stopOnScriptError') ? payload.stopOnScriptError :
                     this.options.stopOnScriptError,
-                abortOnError = _.has(payload, 'abortOnError') ? payload.abortOnError : this.options.abortOnError,
+                abortOnError = abortOnFailure || (_.has(payload, 'abortOnError') ?
```
Comment (shamasis) [inline lib/runner/extensions/event.command.js]: write inline doc

File: lib/backpack/index.js [hunk 2]
```diff
@@ -19,6 +19,10 @@ meetExpectations = function (subject, expectations, defaults) {
 };
 
 module.exports = backpack = {
+    ensure: function (arg) {
```
Comment (shamasis) [inline lib/backpack/index.js]: document and add tests to backpack.ensure please

Comment (shamasis) [inline lib/backpack/index.js]: maybe take in a context too.

File: package.json [hunk 3]
```diff
@@ -22,6 +22,7 @@
   "author": "Postman Labs <help@getpostman.com>",
   "license": "Apache-2.0",
   "dependencies": {
+    "async": "2.0.0",
```
Comment (shamasis) [inline package.json]: it's not removed from dev dependencies

File: lib/runner/extensions/item.command.js [hunk 4]
```diff
@@ -124,6 +124,7 @@ module.exports = {
                         item: item,
                         coords: coords,
                         context: ctxTemplate,
+                        trackFailuresIn: 'tests',
```
Comment (shamasis) [inline lib/runner/extensions/item.command.js]: cleanup

File: package.json (hunk 3 above)
Comment (shamasis) [inline package.json]: cleanup

File: lib/runner/run.js [hunk 5]
```diff
@@ -55,6 +56,8 @@ _.extend(Run.prototype, {
      * @param {Function|Object} callback
      */
     start: function (callback) {
+        var self = this;
```
Comment (shamasis) [inline lib/runner/run.js]: remove self and bind instead

File: lib/runner/run.js [hunk 6]
```diff
@@ -1,4 +1,5 @@
 var _ = require('lodash'),
+    async = require('async'),
```
Comment (shamasis) [inline lib/runner/run.js]: cleanup

File: lib/runner/extensions/event.command.js [hunk 7]
```diff
@@ -111,12 +116,11 @@ module.exports = {
...
             // at this point, the one who queued this event, must ensure that the trigger for it is defined in its
             // 'trigger' interface
-            util.safeCall(this.triggers[_.camelCase('before-' + eventName)], this.triggers, null, cursor,
-                events, item);
+            this.triggers[_.camelCase('before-' + eventName)](null, cursor, events, item);
 
             // with all the event listeners in place, we now iterate on them and execute its scripts. post execution,
             // we accumulate the results in order to be passed on to the event callback trigger.
-            util.asyncEach(events, function (event, next) {
+            async.mapSeries(events, function (event, next) {
```
Comment (shamasis) [inline lib/runner/extensions/event.command.js]: cleanup

File: lib/runner/extensions/event.command.js [hunk 8]
```diff
@@ -91,9 +94,11 @@ module.exports = {
                     payload.trackContext.filter(function (variable) {
                         return _.isObject(payload.context[variable]);
                     }),
+                trackFailuresIn = _.isString(payload.trackFailuresIn) && payload.trackFailuresIn,
```
Comment (shamasis) [inline lib/runner/extensions/event.command.js]: cleanup

File: lib/runner/extensions/event.command.js [hunk 9]
```diff
@@ -72,6 +74,7 @@ module.exports = {
          * @param {Object} [payload.context]
          * @param {Cursor} [payload.coords]
          * @param {Array.<String>} [payload.trackContext]
+         * @param {Array.<String>} [payload.trackFailuresIn]
```
Comment (shamasis) [inline lib/runner/extensions/event.command.js]: cleanup

File: lib/runner/extensions/event.command.js [hunk 10]
```diff
@@ -1,4 +1,6 @@
 var _ = require('lodash'),
+    async = require("async"),
+
```
Comment (shamasis) [inline lib/runner/extensions/event.command.js]: extra whitespace

Comment (czardoz) [inline lib/runner/extensions/event.command.js]: I like to keep internal and external requires separated by an empty line :smile:

**Pull request merged**